src/mixins/soft-crud.repository.mixin.ts

```diff
@@ -1,342 +1,22 @@
-        where &&
-        (where as OrClause<E>).or &&
-        (where as OrClause<E>).or.length > 0
-      ) {
-        where = {
-          and: [
-            {
-              deleted: false,
-            } as Condition<E>,
-            {
-              or: (where as OrClause<E>).or,
-            },
-          ],
-        };
-      } else {
-        where = where ?? {};
-        (where as Condition<E>).deleted = false;
-      }
-
-      // Now call super
-      return super.count(where, options);
-    }
-
-    async delete(entity: E, options?: Options): Promise<void> {
-      // Do soft delete, no hard delete allowed
-      (entity as SoftDeleteEntity).deleted = true;
-      (entity as SoftDeleteEntity).deletedOn = new Date();
-      (entity as SoftDeleteEntity).deletedBy = await this.getUserId(
-        this.getCurrentUser,
-      );
-      return super.update(entity, options);
-    }
-
-    async deleteAll(where?: Where<E>, options?: Options): Promise<Count> {
-      // Do soft delete, no hard delete allowed
-      return this.updateAll(
-        {
-          deleted: true,
-          deletedOn: new Date(),
-          deletedBy: await this.getUserId(this.getCurrentUser),
-        } as DataObject<E>,
-        where,
-        options,
-      );
-    }
-
-    async deleteById(id: ID, options?: Options): Promise<void> {
-      // Do soft delete, no hard delete allowed
-      return super.updateById(
-        id,
-        {
-          deleted: true,
-          deletedOn: new Date(),
-          deletedBy: await this.getUserId(this.getCurrentUser),
-        } as DataObject<E>,
-        options,
-      );
-    }
-
-    /**
-     * Method to perform hard delete of entries. Take caution.
-     * @param entity
-     * @param options
-     */
-    deleteHard(entity: E, options?: Options): Promise<void> {
-      // Do hard delete
-      return super.deleteById(entity.getId(), options);
-    }
-
-    /**
-     * Method to perform hard delete of entries. Take caution.
-     * @param entity
-     * @param options
-     */
-    deleteAllHard(where?: Where<E>, options?: Options): Promise<Count> {
-      // Do hard delete
-      return super.deleteAll(where, options);
-    }
-
-    /**
-     * Method to perform hard delete of entries. Take caution.
-     * @param entity
-     * @param options
-     */
-    deleteByIdHard(id: ID, options?: Options): Promise<void> {
-      // Do hard delete
-      return super.deleteById(id, options);
-    }
-
-    private async getUserId(options?: Options): Promise<string | undefined> {
-      if (!this.getCurrentUser) {
-        return undefined;
-      }
-      let currentUser = await this.getCurrentUser();
-      currentUser = currentUser ?? options?.currentUser;
-      if (!currentUser) {
-        return undefined;
-      }
-      const userId = currentUser.getIdentifier?.() ?? currentUser.id;
-      return userId?.toString();
-    }
-  }
-  return SoftCrudRepository;
+  @extendPrototype(base, SoftCrudRepository)
```

Comment:
Don't we need the opposite of this? I mean, this would override all the changes from the base class na?

Reply:
Ah, my bad we don't need base here (as first layer of extension is already been done using `extends` just below this line). 

Let me update this.

Reply:
Fixed and added comments, please check.